src/System.Management.Automation/engine/SessionStateDriveAPIs.cs

```diff
@@ -508,12 +508,13 @@ private PSDriveInfo GetDrive(string name, bool automount)
 
             if (result == null && automount)
             {
-                result = AutomountBuiltInDrive(name);
-            }
-
-            if (result == null && this == ExecutionContext.TopLevelSessionState)
-            {
-                result = AutomountFileSystemDrive(name);
+                // first try to automount as a file system drive
+                result = AutomountFileSystemDrive(name);

```

Comment:
The old code only created file system drives if the current scope was global.
Will your change create the drive in global scope or module scope?
Personally, I would expect the drive to be in global scope.
I would also like to see tests to cover that scenario either way.